### PR TITLE
fix(PeriphDrivers, Other): Fix RTC clock source selection

### PR DESCRIPTION
**Pull request by mertvatansever**
### Description

This PR solves RTC clock source selection problem for ME30. It adds a function to set clock source and creates wrapper function for MXC_RTC_Init to select clock source for ME30.

### CODE REVIEW

File: Libraries/PeriphDrivers/Source/RTC/rtc_me30.c [hunk 1]
```diff
@@ -65,6 +65,30 @@ int MXC_RTC_SetSubsecondAlarm(uint32_t rssa)
     return MXC_RTC_RevA_SetSubsecondAlarm((mxc_rtc_reva_regs_t *)MXC_RTC, rssa);
 }
 
+int MXC_RTC_SetClockSource(mxc_rtc_clock_t clk_src)
+{
+    uint8_t retval = E_NO_ERROR;
+
+    switch (clk_src) {
+    case MXC_RTC_ERTCO_CLK:
+        MXC_GCR->clkctrl |= MXC_F_GCR_CLKCTRL_ERTCO_EN;
```
Comment (hfakkiz) [inline Libraries/PeriphDrivers/Source/RTC/rtc_me30.c:74]: Please use "MSDK_NO_GPIO_CLK_INIT" macro in here.

File: Libraries/zephyr/MAX/Include/wrap_max32_rtc.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        clk_src = MXC_RTC_INRO_CLK;
+        break;
+    default:
+        return -1;
+    }
+
+    return clk_src;
+}
+
+static inline int WRAP_MXC_RTC_Init(uint32_t sec, uint16_t ssec, uint8_t clock_source)
```
Comment (hfakkiz) [inline Libraries/zephyr/MAX/Include/wrap_max32_rtc.h:49]: Please update wrapper function names as "Wrap_MXC..." to have same name structure with other wrapper files.

File: Libraries/zephyr/MAX/Include/wrap_max32_rtc.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    ret = MXC_RTC_Init(sec, ssec);
+
+    return ret;
+}
+
+#else
+
+static inline int WRAP_MXC_RTC_Init(uint32_t sec, uint16_t ssec, uint8_t clock_source)
+{
+    int ret;
```
Comment (hfakkiz) [inline Libraries/zephyr/MAX/Include/wrap_max32_rtc.h:67]:
```suggestion
    (void)clock_source;
    int ret;
```

Please add this line for unused parameters.

**Review by hfakkiz — CHANGES_REQUESTED** [review]

File: Libraries/PeriphDrivers/Source/RTC/rtc_me30.c [hunk 4]
```diff
@@ -65,6 +65,30 @@ int MXC_RTC_SetSubsecondAlarm(uint32_t rssa)
...
+        MXC_MCR->ctrl &= ~MXC_F_MCR_CTRL_CLKSEL;
+        MXC_MCR->ctrl |= MXC_S_MCR_CTRL_CLKSEL_ERTCO;
```
Comment (hfakkiz) [inline Libraries/PeriphDrivers/Source/RTC/rtc_me30.c:76]:
```suggestion
        MXC_SETFIELD(MXC_MCR->ctrl, MXC_F_MCR_CTRL_CLKSEL, MXC_S_MCR_CTRL_CLKSEL_ERTCO);
```
I think you should use MXC_SETFIELD macro here.

File: Libraries/zephyr/MAX/Include/wrap_max32_rtc.h (hunk 3 above)
Comment (hfakkiz) [reply Libraries/zephyr/MAX/Include/wrap_max32_rtc.h:67]: I checked Zephyr side now and I think adding `if (clock_source != ADI_MAX32_PRPH_CLK_SRC_ERTCO) { return -1; }` line would be nice to handle a bad parameter case.

File: Libraries/PeriphDrivers/Source/RTC/rtc_me30.c (hunk 1 above)
Comment (mertvatansever) [reply Libraries/PeriphDrivers/Source/RTC/rtc_me30.c]: Done, thanks

**Review by hfakkiz — APPROVED** [review]

File: Libraries/PeriphDrivers/Include/MAX32657/rtc.h [hunk 5]
```diff
@@ -75,6 +75,14 @@ typedef enum {
     MXC_RTC_INT_FL_READY = MXC_F_RTC_CTRL_RDY, /**< Timer ready interrupt flag */
 } mxc_rtc_int_fl_t;
 
+/**
+ * @brief     Clock settings
+ */
+typedef enum {
+    MXC_RTC_ERTCO_CLK = 4,
+    MXC_RTC_INRO_CLK = 5,
+} mxc_rtc_clock_t;
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/rtc.h:84]:
```suggestion
typedef enum {
    MXC_RTC_ERTCO_CLK = 0,
    MXC_RTC_INRO_CLK = 1,
    MXC_RTC_EXT_CLK = 2,
} mxc_rtc_clock_t;
```
`wrap_get_clock_source_instance` already does a conversion.

File: Libraries/PeriphDrivers/Source/RTC/rtc_me30.c [hunk 6]
```diff
@@ -65,6 +65,30 @@ int MXC_RTC_SetSubsecondAlarm(uint32_t rssa)
...
+#ifndef MSDK_NO_GPIO_CLK_INIT
+        retval = MXC_SYS_ClockSourceEnable(MXC_SYS_CLOCK_ERTCO);
+#endif // MSDK_NO_GPIO_CLK_INIT
+        MXC_SETFIELD(MXC_MCR->ctrl, MXC_F_MCR_CTRL_CLKSEL, MXC_S_MCR_CTRL_CLKSEL_ERTCO);
+        break;
+
+    case MXC_RTC_INRO_CLK:
+        MXC_SETFIELD(MXC_MCR->ctrl, MXC_F_MCR_CTRL_CLKSEL, MXC_S_MCR_CTRL_CLKSEL_INRO_DIV4);
+        break;
+
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Source/RTC/rtc_me30.c:83]: You could add a case for MXC_RTC_EXT_CLK and return E_NOT_SUPPORTED.

**Review by ttmut — CHANGES_REQUESTED** [review]

**Review by ttmut — APPROVED** [review]

**Pull request merged**